### DISCUSSION TITLE
Just want to thank you for your dedication. This utility rocks!!!!

### DISCUSSION THREAD

**Discussion by csc-chicken** [ValShaped/rwfus#38] [General]
It is one of the best tool for steamos I think. Thank you!!

**Comment by ValShaped** [comment]
Thanks for using it! ❤️

**Comment by fndpena** [comment]
I agree, it truly expands the deck potential, it's a must have for all deck users that rely on desktop mode.

Btw, I have a question for you @ValShaped.

What happens if an existing package (from steamOS) is updated using rwfus overlay?
I know the new version is gonna be installed properly, but what if I wanna rollback to the original (steamOS) package version?
If I run `sudo -R package-name` only the overlayed version is removed? but the original packlage will still be there?

**Reply by ValShaped** [reply]
If you uninstall a package inside the overlay, the contents of the package will be given deletion-marker files in the overlay 's upper directory (which, for Rwfus, is in a subdirectory of /opt/rwfus/mount). Even though the original files are present, they will appear to be deleted as long as the overlay is mounted.

As much as Rwfus makes it easy to install software, it makes it very difficult to uninstall individual packages.

Additionally, *every* package in the overlay is carried, even if the base image has a newer version. This is why installing `glibc` in the overlay is explicitly disallowed (it can and will break everything.) I wrote Rwfus with the goal of keeping Yakuake installed on my Deck, and I'm happy to see what other people do with it, but I'll be the first to admit it has a lot of unmarked pitfalls.

If you want that sort of piecemeal addition and deletion, I'd recommend getting familiar with the Nix package manager—it takes a totally different, more compatible approach to dependency management.

**Reply by fndpena** [reply]
Ok so... I'm asking this because I updated a specific package: gamescope
It's not causing any issues at all, but I was wondering what would happen if I run `sudo pacman -R` on it.

If I understood correctly, if I do that, when the overlay is mounted the package will not be listed when I `sudo pacman -Q`, even though the original package is still there.

In that case, is there a way to restore that? Maybe re-mounting the overlay?

**Reply by ValShaped** [reply]
Unmounting the overlay will restore it, re-mounting the overlay will remove it again. You'll have to remove the deletion markers from the overlay upperdirs if you want to keep the overlay around.

If you don't care about the state of the overlay, you can always delete it and make a new one (the Rwfus manager utility has command line arguments for backing up and restoring the overlay disk image. I believe `rwfus -h` will show the full list of options?)
